metaflow/plugins/kubernetes/kubernetes.py

```diff
@@ -280,18 +280,7 @@ def create_job(
         for name, value in env.items():
             job.environment_variable(name, value)
 
-        annotations = {
-            "metaflow/user": user,
-            "metaflow/flow_name": flow_name,
-        }
-        if current.get("project_name"):
-            annotations.update(
-                {
-                    "metaflow/project_name": current.project_name,
-                    "metaflow/branch_name": current.branch_name,
-                    "metaflow/project_flow_name": current.project_flow_name,
-                }
-            )
+        annotations = self._get_annotations(user, flow_name, current)
```

Comment:
Backpedaling on my previous comments, as I'm a bit unsure whether this is cleaner than the original approach where `_get_annotations` only parses the env. Now we need to pass in a bunch of additional details to be able to generate the system annotations as well.

Not a dealbreaker for shipping this, I leave it up to you which approach you find better :)

Reply:
I think I do prefer the original approach. It makes it more difficult to follow the logic of the annotations with the system level annotations being put in the `_get_annotations` function. 

I reverted the changes to the function